File: backend/src/main/java/com/example/demo/Entity/R_FD.java
package com.example.demo.Entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "r_fd" , uniqueConstraints = {@UniqueConstraint(columnNames = {"dipendente_id", "giornata_ferie_id"})} )
public class R_FD {

    @Id
    @GeneratedValue
    private long id;

    @ManyToOne
    @JoinColumn(name = "dipendente_id")
    private Dipendente dipendente;

    @ManyToOne
    @JoinColumn(name = "giornata_ferie_id")
    private GiornataFeriale giornataFeriale;


}

File: frontend/src/app/model/rfd.ts
import { Dipendente } from "./Dipendente";
import { GiornataFeriale } from "./GiornateFerie";

export interface rfd{
    id:number,
    dipendente:Dipendente,
    giornataFeriale:GiornataFeriale
}

export class rfd implements rfd {
    id!:number;
    dipendente:Dipendente;
    giornataFeriale:GiornataFeriale;

    constructor(dip:Dipendente, gf:GiornataFeriale){
        this.dipendente = dip;
        this.giornataFeriale = gf;
    }
}